src/BoardEvaluator.java

```diff
@@ -0,0 +1,13 @@
+
+public interface BoardEvaluator {
```

Comment:
I like having an interface for BoardEvaluator. I'd just say split it out into it's own .java file. :)